source/docs/naming_javascript.md

```diff
@@ -1,16 +1,38 @@
-title: Naming System (ENS) Usage in Javacript
+title: Naming System (ENS) Usage in JavaScript
 ---
 
+In order to improve the user experience of decentralized applications that deal with smart contracts or Ethereum addresses in general, it's a good practice to take advantage of the [Ethereum Name Service](https://ens.domains/) and its registered, human-readable names. Embark provides APIs to either resolve a given domain to its corresponding address, or lookup a registered name by a given address.
+
+Let's take a look at how this is done.
+
 ### Resolving a Domain to an address
 
-<pre><code class="javascript">EmbarkJS.Names.resolve("ethereum.eth").then(function(address) {
+Resolving a domain to its corresponding address is done by calling `EmbarkJS.Names.resolve`, which takes the domain in question as first argument and a callback function that has access to the resolve address, as a second argument.and turns a promise that resolves with an address or an error message.
```

Comment:
`to the resolve address, as a second argument.and turns` -> `to the resolved address as a second argument, and returns`

Reply:
Although you do mention the promise below. This is a bit repetitive.

Reply:
Oh shoot. This is actually a bug. I didn't mean to mention the Promise at this point yet haha.

This is because I first had it the other way around and then decided to first introduce the callback APi and then the Promise API. Will update this.